From: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Date: Wed, 25 Apr 2012 01:46:28 +0000
Subject: [PATCH 46/58] be2net: Fix ethtool self test for Lancer

commit a70473949698c1c46e5523e1810a0a0f904a1fd2 upstream.

Lancer does not support DDR self test. Fix ethtool self test by
skipping this test for Lancer.

Signed-off-by: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/emulex/benet/be_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c
===================================================================
--- linux-3.2.46.orig/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 19:21:49.000000000 +0000
+++ linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 19:21:53.000000000 +0000
@@ -793,7 +793,7 @@
 		}
 	}
 
-	if (be_test_ddr_dma(adapter) != 0) {
+	if (!lancer_chip(adapter) && be_test_ddr_dma(adapter) != 0) {
 		data[3] = 1;
 		test->flags |= ETH_TEST_FL_FAILED;
 	}
